### ISSUE TITLE
[Bug]:  fetch failed for source Apps by Abfall+

### ISSUE THREAD

**Issue by Barneybaer84** [mampfes/hacs_waste_collection_schedule#1411]
### I Have A Problem With:

A specific source, The integration in general

### What's Your Problem

Bekomme es nicht zum Laufen. Immer Fehler bei der Einrichtung.

### Source (if relevant)

 de.abfallwecker

### Logs

```Shell
` Logger: waste_collection_schedule.source_shell
Source: custom_components/waste_collection_schedule/waste_collection_schedule/source_shell.py:136
Integration: waste_collection_schedule (documentation)
First occurred: 21:14:32 (1 occurrences)
Last logged: 21:14:32
fetch failed for source Apps by Abfall+: Traceback (most recent call last): File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source_shell.py", line 134, in fetch entries = self._source.fetch() ^^^^^^^^^^^^^^^^^^^^ File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source/app_abfallplus_de.py", line 109, in fetch for d in self._app.generate_calendar(): ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/service/AppAbfallplusDe.py", line 676, in generate_calendar self.select_bundesland() File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/service/AppAbfallplusDe.py", line 401, in select_bundesland for bundesland in self.get_bundeslaender(): ^^^^^^^^^^^^^^^^^^^^^^^^ File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/service/AppAbfallplusDe.py", line 393, in get_bundeslaender "name": a[1], ~^^^ IndexError: list index out of range `
```

### Relevant Configuration

```YAML
waste_collection_schedule:
    sources:
    - name: app_abfallplus_de
      args:
        app_id: de.abfallwecker
        bundesland: Sachsen
        landkreis: Landkreis Nordsachsen
        city: xxx
        strasse: xxx
        hnr: x
```

### Checklist Source Error

- [X] Use the example parameters for your source (often available in the documentation) (don't forget to restart Home Assistant after changing the configuration)
- [X] Checked that the website of your service provider is still working
- [X] Tested my attributes on the service provider website (if possible)
- [X] I have tested with the latest version of the integration (master) (for HACS in the 3 dot menu of the integration click on "Redownload" and choose master as version)

### Checklist Sensor Error

- [X] Checked in the Home Assistant Calendar tab if the event names match the types names (if types argument is used)

### Required

- [X] I have searched past (closed AND opened) issues to see if this bug has already been reported, and it hasn't been.
- [X] I understand that people give their precious time for free, and thus I've done my very best to make this problem as easy as possible to investigate.

**Comment by 5ila5** [comment]
There is deffenetly something wrong with handling de.abfallwecker it somehow messes up Bundesland selection.
Most of the supported apps do not require this step. I will look into it.

**Comment by 5ila5** [comment]
Fixed in above #1414.
Fix will be part of release 1.44.0 or is available now as master (git) version (HACS: 3-dot menu -> redownload -> change version to master)
